### PR TITLE
Add support for gauge chart type

### PR DESCRIPTION
**Pull request by rob-duplock**
- Added support for gauge chart type
- Refactored from existing PR with improvements / coding standard fixes

### CODE REVIEW

File: js/jquery.dvfCharts.js [hunk 1]
```diff
@@ -280,6 +281,48 @@
       return this;
     },
 
+    parseGaugeOptions: function () {
```
Comment (jez500) [inline js/jquery.dvfCharts.js]: I know this is just following suit with Jorges zero comments in Javascript :P but I feel every method should have at least one comment describing its purpose and in cases like this a link to the c3js reference.

eg 
```
/**
 * Parse user settings into object suitable for use with Gauge Chart.
 *
 * @example https://c3js.org/samples/chart_gauge.html
 * @see https://c3js.org/reference.html#gauge-label-show
 */
parseGaugeOptions: function () {
  ...
}
```

**Review by jez500 — APPROVED** [review]
Looks good apart from my comment gripe :)

**Comment by jez500** [comment]
Have also asked @rob-duplock to ensure the split field works with this so it doesn't have to solely rely on totals

**Review by jorgegc — APPROVED** [review]
Looking good @rob-duplock. Nice work!

**Review by jorgegc — CHANGES_REQUESTED** [review]
@rob-duplock and I had a chat about a small bug we found with split fields and gauge charts. He will update the PR shortly.

**Comment by rob-duplock** [comment]
I have added the split field bugfix.
It should also be noted the best gauge chart experience relies on having the[ C3 and D3 libraries updated to the latest version](https://github.com/govCMS/dvf/pull/24).

**Review by jorgegc — APPROVED** [review]
Thanks @rob-duplock. Looks good!

**Review by tobybellwood — APPROVED** [review]
I'm happy for @jorgegc to merge.

**Pull request merged**